packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,360 @@
+if (process.argv.length != 3 && process.argv.length != 4) {
+  console.error(
+    'Usage: ' +
+      process.argv[0] +
+      ' ' +
+      process.argv[1] +
+      ' path/to/rippled [path/to/pipe/to]',
+  )
+  process.exit(1)
+}
+
+////////////////////////////////////////////////////////////////////////
+//  Get all necessary files from rippled
+////////////////////////////////////////////////////////////////////////
+const path = require('path')
+
+const fs = require('fs')
+function readFile(filename) {
+  try {
+    return fs.readFileSync(filename).toString('utf-8')
+  } catch (err) {
+    console.error(`Error reading file ${filename}:`, err.message)
+    process.exit(1)
+  }
+}
+
+const sfieldHeaderFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/SField.h'),
+)
+const sfieldMacroFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/detail/sfields.macro'),
+)
+const ledgerFormatsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/ledger_entries.macro',
+  ),
+)
+const terFile = readFile(
+  path.join(process.argv[2], '/include/xrpl/protocol/TER.h'),
+)
+const transactionsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/transactions.macro',
+  ),
+)
+
+const capitalizationExceptions = {
+  NFTOKEN: 'NFToken',
+  URITOKEN: 'URIToken',
+  URI: 'URI',
+  UNL: 'UNL',
+  XCHAIN: 'XChain',
+  DID: 'DID',
+  ID: 'ID',
+  AMM: 'AMM',
```

Comment:
This dictionary can be stripped down to the following elements:
```
const capitalizationExceptions = {
  NFTOKEN: 'NFToken',
  URITOKEN: 'URIToken',
  XCHAIN: 'XChain',
}
```

If you prefer, feel free to cherry-pick this commit [47a1e62b5ee2cfd2ed17d23dd7cb8adcbddf21b7](https://github.com/XRPLF/xrpl.js/commit/47a1e62b5ee2cfd2ed17d23dd7cb8adcbddf21b7). I have used the latest tip of `develop` branch from rippled.